lnutil/msglib.go

```diff
@@ -1,47 +1,770 @@
+	case MSGID_REV:
+		return NewRevMsgFromBytes(b, peerid)
+
+	/*
+		case MSGID_FWDMSG:
+		case MSGID_FWDAUTHREQ:
+
+		case MSGID_SELFPUSH:
+	*/
+
+	case MSGID_WATCH_DESC:
+		return NewWatchDescMsgFromBytes(b, peerid)
+	case MSGID_WATCH_COMMSG:
+		return NewComMsgFromBytes(b, peerid)
+	/*
+		case MSGID_WATCH_DELETE:
+	*/
+
+	default:
+		return nil, fmt.Errorf("Unknown message of type %d ", msgType)
+	}
+}
+
+//----------
+
+//text message
+type ChatMsg struct {
 	PeerIdx uint32
-	ChanIdx uint32 // optional, may be 0
-	MsgType uint8
-	Data    []byte
+	Text    string
 }
 
-const (
-	MSGID_POINTREQ  = 0x30
-	MSGID_POINTRESP = 0x31
-	MSGID_CHANDESC  = 0x32
-	MSGID_CHANACK   = 0x33
-	MSGID_SIGPROOF  = 0x34
+func NewChatMsg(peerid uint32, text string) *ChatMsg {
+	t := new(ChatMsg)
+	t.PeerIdx = peerid
+	t.Text = text
+	return t
+}
 
-	MSGID_CLOSEREQ  = 0x40 // close channel
-	MSGID_CLOSERESP = 0x41
+func NewChatMsgFromBytes(b []byte, peerid uint32) (*ChatMsg, error) {
+	c := new(ChatMsg)
+	c.PeerIdx = peerid
+	b = b[1:]
+	c.Text = string(b)
 
-	MSGID_TEXTCHAT = 0x60 // send a text message
+	return c, nil // no way to error
+}
 
-	MSGID_DELTASIG  = 0x70 // pushing funds in channel; request to send
-	MSGID_SIGREV    = 0x72 // pulling funds; signing new state and revoking old
-	MSGID_GAPSIGREV = 0x73 // resolving collision
-	MSGID_REV       = 0x74 // pushing funds; revoking previous channel state
+func (self ChatMsg) Bytes() []byte {
+	var msg []byte
+	msg = append(msg, self.MsgType())
+	msg = append(msg, []byte(self.Text)...)
+	return msg
+}
 
-	MSGID_FWDMSG     = 0x20
-	MSGID_FWDAUTHREQ = 0x21
+func (self ChatMsg) Peer() uint32   { return self.PeerIdx }
+func (self ChatMsg) MsgType() uint8 { return MSGID_TEXTCHAT }
 
-	MSGID_SELFPUSH = 0x80
-)
+//----------
+
+//message with no information, just shows a point is requested
+type PointReqMsg struct {
+	PeerIdx uint32
+}
+
+func NewPointReqMsg(peerid uint32) *PointReqMsg {
+	p := new(PointReqMsg)
+	p.PeerIdx = peerid
+	return p
+}
+
+func NewPointReqMsgFromBytes(peerid uint32) (*PointReqMsg, error) {
+	return NewPointReqMsg(peerid), nil // no way to have error
+}
+
+func (self PointReqMsg) Bytes() []byte {
+	var msg []byte
+	msg = append(msg, self.MsgType())
+	return msg
+}
+
+func (self PointReqMsg) Peer() uint32   { return self.PeerIdx }
+func (self PointReqMsg) MsgType() uint8 { return MSGID_POINTREQ }
+
+//message to be used for reply to point request
+type PointRespMsg struct {
+	PeerIdx    uint32
+	ChannelPub [33]byte
+	RefundPub  [33]byte
+	HAKDbase   [33]byte
+}
+
+func NewPointRespMsg(peerid uint32, chanpub [33]byte, refundpub [33]byte, HAKD [33]byte) *PointRespMsg {
+	pr := new(PointRespMsg)
+	pr.PeerIdx = peerid
+	pr.ChannelPub = chanpub
+	pr.RefundPub = refundpub
+	pr.HAKDbase = HAKD
+	return pr
+}
+
+func NewPointRespMsgFromBytes(b []byte, peerid uint32) (*PointRespMsg, error) {
+	pm := new(PointRespMsg)
+	b = b[1:] // get rid of messageType
+	pm.PeerIdx = peerid
+
+	if len(b) != 99 {
```

Comment:
similarly here, move the len check up to the top, before the truncation. (set to 100 I guess)